### PR TITLE
Es6 features

### PR DESCRIPTION
**Pull request by basem909**
## Description

- This project aims to create a javascript-based project to develop a UI site that takes data from the user and shows it on the screen and enables the user to remove data as well.
- The awesome book is a list of books the user could add to it and remove books from it and the entire data is saved on the local storage of the user.
- The project follows ES6 best practices and contains luxon library from which it retrieved time and date.

## Built With

- HTML
- CSS
- Javascript
- node-modules
- lighthouse
- hintweb
- stylint
- eslint
- luxon

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+const library = new Library();
+
+function loadBooks() {
+  app.innerHTML = '';
+  const listElement = document.createRange().createContextualFragment(list(library));
+  app.append(listElement);
+  const booksList = document.querySelector('#book-list');
+  booksList.innerHTML = '';
+  library.load(booksList);
+}
```
Comment (keroloussamy) [inline index.js]: - [ ] Kindly refactor your methods and functions to arrow functions. as required from the requirements.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  const bookAuthor = document.querySelector('#book-author');
+
+  bookForm.addEventListener('submit', (e) => {
+    e.preventDefault();
+    const book = new Book(bookTitle.value, bookAuthor.value);
+    library.addBook(book);
+    bookTitle.value = '';
+    bookAuthor.value = '';
+  });
+}
```
Comment (keroloussamy) [inline index.js]: - [ ] Kindly refactor your methods and functions to arrow functions. as required from the requirements.

File: modules/Library.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        this.books.splice(i, 1);
+        booksList.innerHTML = '';
+        for (let i = 0; i < this.books.length; i += 1) {
+          insertBookIntoDom(this.books[i], booksList);
+        }
+        this.#saveIntoStorage();
+        break;
+      }
+    }
+  }
```
Comment (keroloussamy) [inline modules/Library.js]: - [ ] Kindly refactor your methods and functions to arrow functions. as required from the requirements.

File: modules/Library.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  load(bookList) {
+    const booksData = JSON.parse(localStorage.getItem('books'));
+    if (booksData) {
+      this.books = [];
+      for (let i = 0; i < booksData.length; i += 1) {
+        this.books.push(booksData[i]);
+        insertBookIntoDom(booksData[i], bookList);
+      }
+    }
```
Comment (keroloussamy) [inline modules/Library.js]: - [ ] Kindly refactor your methods and functions to arrow functions. as required from the requirements.

**Review by keroloussamy — CHANGES_REQUESTED** [review]
Hi @basem909, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️

## Required Changes ♻️
_Check the comments under the review._




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/Library.js (hunk 4 above)
Comment (basem909) [reply modules/Library.js]: Hi @keroloussamy 
I had my code approved while you have asked to make changes.
Would you please check out my new changes and make sure that the project is ready to merge?!

Comment (keroloussamy) [reply modules/Library.js]: Hello @basem909, Yes Your project is complete now, Good Job.

**Pull request merged**